<launch>
  <arg name="sample" default=""/>

  <!-- For infield correction -->
  <arg name="operation" default=""/>

  <!-- For hand-eye calibration -->
  <arg name="configuration" default=""/>
  <arg name="marker_ids" default="[-1]"/> <!-- Empty integer arrays cannot be specified, using -1 as a placeholder. -->
  <arg name="working_directory" default=""/>

  <node pkg="zivid_camera" exec="zivid_camera" name="zivid_camera" />
  <node pkg="zivid_samples" exec="$(var sample)" name="zivid_sample">
    <param name="operation" value="$(var operation)"/>
    <param name="configuration" value="$(var configuration)"/>
    <param name="marker_ids" value="$(var marker_ids)"/>
    <param name="working_directory" value="$(var working_directory)"/>
  </node>
</launch>
